File: services/blog-service/src/main/java/com/platform/blog/domain/entity/Tag.java
package com.platform.blog.domain.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.platform.common.core.domain.entity.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 标签实体
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("biz_tag")
public class Tag extends BaseEntity {

    /**
     * 标签名称
     */
    private String name;

    /**
     * 标签别名（URL友好）
     */
    private String slug;

    /**
     * 描述
     */
    private String description;

    /**
     * 颜色
     */
    private String color;

    /**
     * 排序
     */
    private Integer sort;

    /**
     * 状态: 1-启用, 0-禁用
     */
    private Integer status;
}

File: frontend/blog-web/src/api/tag.ts
import request from '@/utils/request'
import type { ApiResult } from '@/types/api'

export interface TagVO {
  id: number
  name: string
  slug: string
  description: string
  color: string
  sort: number
  status: number
}

export function getAllTags() {
  return request.get<ApiResult<TagVO[]>>('/blog/tags')
}
